drivers/mdio/mdio_xilinx_axienet.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Xilinx AXI 1G / 2.5G Ethernet Subsystem
+ *
+ * Copyright(c) 2024, CISPA Helmholtz Center for Information Security
+ * SPDX - License - Identifier : Apache-2.0
+ */
+#define LOG_MODULE_NAME eth_xilinx_axienet_mdio
+#define LOG_LEVEL       CONFIG_ETHERNET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <sys/types.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/sys/barrier.h>
+
+#define XILINX_AXIENET_MDIO_SETUP_REG_OFFSET                  0x00000500
+#define XILINX_AXIENET_MDIO_SETUP_REG_MDIO_DISABLE_MASK       0x0
+#define XILINX_AXIENET_MDIO_SETUP_REG_MDIO_ENABLE_MASK        (1 << 6)
+#define XILINX_AXIENET_MDIO_SETUP_REG_MDIO_CLOCK_DIVIDER_MASK ((1 << 6) - 1)
+#define XILINX_AXIENET_MDIO_CONTROL_REG_OFFSET                0x00000504
+#define XILINX_AXIENET_MDIO_CONTROL_REG_MASK_READY            (1 << 7)
+#define XILINX_AXIENET_MDIO_CONTROL_REG_SHIFT_PHYADDR         24
+#define XILINX_AXIENET_MDIO_CONTROL_REG_SHIFT_REGADDR         16
+#define XILINX_AXIENET_MDIO_CONTROL_REG_SHIFT_TXOP            14
+#define XILINX_AXIENET_MDIO_CONTROL_REG_MASK_READ             0x00008000
+#define XILINX_AXIENET_MDIO_CONTROL_REG_MASK_WRITE            0x00004000
+#define XILINX_AXIENET_MDIO_CONTROL_REG_MASK_INITIATE         (1 << 11)
+#define XILINX_AXIENET_MDIO_WRITE_DATA_REG_OFFSET             0x00000508
+#define XILINX_AXIENET_MDIO_READ_DATA_REG_OFFSET              0x0000050C
+#define XILINX_AXIENET_MDIO_READ_DATA_REG_DATA_MASK           0x0000ffff
```

Comment:
you could consider using the util_macro.h to make these definitions more easily readable